internal/resource/config/password_credential_validators_resource.go

```diff
@@ -0,0 +1,649 @@
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type passwordCredentialValidatorsResourceModel struct {
+	AttributeContract   types.Object `tfsdk:"attribute_contract"`
+	Id                  types.String `tfsdk:"id"`
+	Name                types.String `tfsdk:"name"`
+	PluginDescriptorRef types.Object `tfsdk:"plugin_descriptor_ref"`
+	ParentRef           types.Object `tfsdk:"parent_ref"`
+	Configuration       types.Object `tfsdk:"configuration"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *passwordCredentialValidatorsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages Password Credential Validators",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the plugin instance. The ID cannot be modified once the instance is created. Note: Ignored when specifying a connection's adapter override.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile("^[a-zA-Z0-9_]{1,32}$"),
+						"The plugin ID must be less than 33 characters, contain no spaces, and be alphanumeric.",
+					),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The plugin instance name. The name can be modified once the instance is created. Note: Ignored when specifying a connection's adapter override.",
+				Required:    true,
+			},
+			"plugin_descriptor_ref": schema.SingleNestedAttribute{
+				Description: "Reference to the plugin descriptor for this instance. The plugin descriptor cannot be modified once the instance is created. Note: Ignored when specifying a connection's adapter override.",
+				Required:    true,
+				Attributes: map[string]schema.Attribute{
+					"id": schema.StringAttribute{
+						Description: "The ID of the resource.",
+						Required:    true,
+					},
+					"location": schema.StringAttribute{
+						Description: "A read-only URL that references the resource. If the resource is not currently URL-accessible, this property will be null.",
+						Computed:    true,
+						Optional:    false,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+			},
+			"parent_ref": schema.SingleNestedAttribute{
+				Description: "The reference to this plugin's parent instance. The parent reference is only accepted if the plugin type supports parent instances. Note: This parent reference is required if this plugin instance is used as an overriding plugin (e.g. connection adapter overrides)",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
```

Comment:
Seems like we might want a utility method for all these ref attributes. Could save a little repetition in our schemas.